src/sm/servicemanager/servicemanager.cpp

```diff
@@ -227,4 +578,28 @@ RetWithError<StaticString<cFilePathLen>> ServiceManager::DigestToPath(const Stri
     return FS::JoinPath(imagePath, cImageBlobsFolder, digestList[0], digestList[1]);
 }
 
+RetWithError<StaticString<ServiceManager::cAllocatorItemLen>> ServiceManager::FormatAllocatorItemID(
+    const ServiceData& service)
+{
+    StaticString<cAllocatorItemLen> id;
+
+    if (auto err = id.Format(cAllocatorItemFormat, service.mServiceID, service.mVersion); !err.IsNone()) {
```

Comment:
@MykolaSolyanko  indicates problem with this function. Could you check?

Reply:
Passed format string is a const char*
static constexpr auto cAllocatorItemFormat = "%s_%s"

As per Mykola's observation, it has an issue when String is  passed and works fine with const char*